public/API/API_GetUserGameLeaderboards.php

```diff
@@ -0,0 +1,124 @@
+ *    object     UserEntry                  details of the requested user's leaderboard entry
+ *     object      [value]
+ *      string     User                     username
+ *      int        Score                    raw value score
+ *      string     FormattedScore           formatted string value of score
+ *      int        Rank                     user's leaderboard rank
+ *      string     DateUpdated              an ISO8601 timestamp string for when the entry was updated
+ */
+
+use App\Models\Game;
+use App\Models\LeaderboardEntry;
+use App\Models\User;
+use App\Platform\Enums\ValueFormat;
+use App\Support\Rules\CtypeAlnum;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Facades\Validator;
+
+$input = Validator::validate(Arr::wrap(request()->query()), [
+    'i' => ['required', 'min:1'],
+    'u' => ['required', 'min:2', 'max:20', new CtypeAlnum()],
+    'o' => ['sometimes', 'integer', 'min:0', 'nullable'],
+    'c' => ['sometimes', 'integer', 'min:1', 'max:500', 'nullable'],
+]);
+
+$offset = $input['o'] ?? 0;
+$count = $input['c'] ?? 200;
+
+$user = User::firstWhere('User', request()->query('u'));
+if (!$user) {
+    return response()->json(['User not found'], 404);
+}
+
+$game = Game::firstWhere("ID", request()->query('i'));
+if (!$game) {
+    return response()->json(['Game not found'], 404);
+}
+
+if ($game->leaderboards()->count() === 0) {
+    return response()->json(['Game has no leaderboards'], 422);
+}
+
+$userLeaderboardEntriesCount = LeaderboardEntry::where('user_id', $user->id)
+    ->whereIn('leaderboard_id', function ($query) use ($game) {
+        $query->select('ID')
+              ->from('LeaderboardDef')
+              ->where('GameID', $game->ID);
+    })
+    ->whereNull('deleted_at')
+    ->count();
+
+if ($userLeaderboardEntriesCount === 0) {
+    return response()->json(['User has no leaderboards on this game'], 422);
+}
+
+$leaderboardEntries = LeaderboardEntry::select('leaderboard_entries.*')
+    ->addSelect([
+        'calculated_rank' => LeaderboardEntry::from('leaderboard_entries as entries_bis')
+            ->join('LeaderboardDef as leaderboardDefBis', 'entries_bis.leaderboard_id', '=', 'leaderboardDefBis.ID')
+            ->whereColumn('entries_bis.leaderboard_id', 'leaderboard_entries.leaderboard_id')
+            ->whereNull('entries_bis.deleted_at')
+            ->whereNull('leaderboardDefBis.deleted_at')
+            ->where(function ($query) {
+                $query->where(function ($q) {
+                    $q->where('leaderboardDefBis.LowerIsBetter', 1)
+                      ->whereColumn('entries_bis.score', '<', 'leaderboard_entries.score');
+                })->orWhere(function ($q) {
+                    $q->where('leaderboardDefBis.LowerIsBetter', 0)
+                      ->whereColumn('entries_bis.score', '>', 'leaderboard_entries.score');
+                });
+            })
+            ->selectRaw('COUNT(*) + 1'),
```

Comment:
I have a feeling that this isn't excluding unranked users.

Reply:
Most likely. Although, I'll need some details in order to confirm my changes are good :)

How does the unranked works? Not null is because they don't want to appear on leaderboards? What's the difference between unranked and untracked?

Thanks in advance

Reply:
Unranked and untracked are synonymous - we plan on ultimately dropping `Untracked` and fully moving over to simply using `unranked_at`, but this migration hasn't happened yet.

Reply:
It looks like the rank calculation needs to filter out deleted entries:
```
{"ID":85289,"RankAsc":true,"Title":"Tutorial Speedster","Description":"Complete the tutorial in the quickest time","Format":"TIMESECS","UserEntry":{"User":"Jamiras","Score":1004,"FormattedScore":"16:44","Rank":16,"DateUpdated":"2024-09-16T00:35:37+00:00"}}]}
```
![image](https://github.com/user-attachments/assets/8a1ab465-d8c8-4681-8d9d-a61bbca6fab6)

![image](https://github.com/user-attachments/assets/c08acabd-91ad-4990-98f4-787b8d1526b0)

```
| 123263 | Nightfire391  |   534 | 2024-07-28 11:37:13 |
| 316263 | Roninx2       |   544 | NULL                |
``

Reply:
Looks like the addSelect its not doing it by default while the regular select does. I added the conditionals to do this again and also covered the results with a test. Feel free to pull changes and give it another try :)